### PR TITLE
[P4-2304] Newly specced events (part 1)

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

P4-2304

### What?

I have added/removed/altered:

- [x] Adds JourneyPersonBoardsVehicle event
- [x] Adds MoveOperationHMCTS event
- [x] Adds PerCourtExcessiveDelayNotDueToSupplier event
- [x] Adds PerCourtAllDocumentationProvidedToSupplier event
- [x] Adds PerCourtReturnToCustodyAreaFromDock event
- [x] Adds PerCourtCellShareRiskAssessment event
- [x] Adds PerCourtAssignCellInCustody event
- [x] Adds PerCourtReadyInCustody event
- [x] Plumbs together PersonEscortRecord as an eventable

### Why?

I am doing this because:

- These are needed by suppliers and the frontend for the generic event api

### Have you? (optional)

- [x] The event api wiki docs have been updated to include these

### Deployment risks (optional)

- These are not currently in use/plumbed in by suppliers so this is very low risk.

### CODE REVIEW

File: app/models/generic_event/per_court_cell_share_risk_assessment.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    validates :location_id, presence: true
+
+    def location_id=(location_id)
+      details['location_id'] = location_id
+    end
+
+    def location_id
+      details['location_id']
+    end
+  end
```
Comment (smoothcontract) [inline app/models/generic_event/per_court_cell_share_risk_assessment.rb:14]: This looks like a good candidate for extracting to a shared concern perhaps - I see it used in a couple of events?

File: spec/models/generic_event/journey_admit_through_outer_gate_spec.rb [hunk 2]
```diff
@@ -2,5 +2,6 @@
   subject(:generic_event) { build(:event_journey_admit_through_outer_gate) }
 
   it { is_expected.to validate_inclusion_of(:eventable_type).in_array(%w[Journey]) }
-  it { is_expected.to validate_presence_of(:vehicle_reg) }
+
+  it_behaves_like 'an event that specifies a vehicle registration'
```
Comment (smoothcontract) [inline spec/models/generic_event/journey_admit_through_outer_gate_spec.rb:6]: 👍

**Review by smoothcontract — COMMENTED** [review]
Nice cleanup - I like the refactoring here to DRY up the implementation as more events are added (and there are so many!). Not sure if you want to introduce a pattern for common attributes across different events, but without validation as these seem to be optional in some cases (e.g. `location_id`) ? Maybe also worth moving the various concerns to a dedicated `events` module just for clarity as I doubt these would be useful outside of the events use case.

File: app/models/generic_event/per_court_cell_share_risk_assessment.rb (hunk 1 above)
Comment (willfish) [reply app/models/generic_event/per_court_cell_share_risk_assessment.rb:14]: Very good idea 👍 .

**Review by smoothcontract — APPROVED** [review]
Discussed offline, moving further DRYing up to a separate PR for clarity, so this is good to go 👍

**Pull request merged**